beacon-chain/state/state-native/getters_state.go

```diff
@@ -324,26 +345,46 @@ func (b *BeaconState) ToProto() interface{} {
 
 // StateRoots kept track of in the beacon state.
 func (b *BeaconState) StateRoots() [][]byte {
-	if b.stateRoots == nil {
-		return nil
-	}
-
 	b.lock.RLock()
 	defer b.lock.RUnlock()
 
-	return b.stateRoots.Slice()
+	roots := b.stateRootsVal()
+	if roots == nil {
+		return nil
+	}
+	return roots.Slice()
+}
+
+func (b *BeaconState) stateRootsVal() customtypes.StateRoots {
+	if features.Get().EnableExperimentalState {
+		if b.stateRootsMultiValue == nil {
+			return nil
+		}
+		return b.stateRootsMultiValue.Value(b)
+	}
+	return b.stateRoots
 }
 
 // StateRootAtIndex retrieves a specific state root based on an
 // input index value.
 func (b *BeaconState) StateRootAtIndex(idx uint64) ([]byte, error) {
```

Comment:
Why does this function return a slice instead of a root?

Reply:
I think originally state roots were kept as `[][]byte`, so this was the easiest thing to return. Same point good be made about block roots and randao mixes. Definitely worth changing, but maybe not in this PR.